soc/espressif/esp32/soc.c

```diff
@@ -185,3 +132,8 @@ void sys_arch_reboot(int type)
 {
 	esp_restart_noos();
 }
+
+#if defined(CONFIG_SOC_ENABLE_APPCPU) && !defined(CONFIG_MCUBOOT)
+extern int esp_appcpu_init(void);
+SYS_INIT(esp_appcpu_init, APPLICATION, 50);
```

Comment:
You could use some of the available priorities such as `KERNEL_INIT_PRIORITY_DEVICE` or `APPLICATION_INIT_PRIORITY` instead of the hard-coded one.